logger_test.go

```diff
@@ -579,6 +579,29 @@ func TestLoggerCustomOnFatal(t *testing.T) {
 	}
 }
 
+type customWriteHook struct {
+	called bool
+}
+
+func (h *customWriteHook) OnWrite(_ *zapcore.CheckedEntry, _ []Field) {
+	h.called = true
+}
+
+func TestLoggerWithFatalHook(t *testing.T) {
+	h := &customWriteHook{}
+	withLogger(t, InfoLevel, opts(WithFatalHook(h)), func(logger *Logger, logs *observer.ObservedLogs) {
+		recovered := make(chan interface{})
+		go func() {
+			defer func() {
+				recovered <- recover()
+			}()
+			logger.Fatal("")
+		}()
+		<-recovered
```

Comment:
The goroutine or channel isn't necessary.
The logger.Fatal won't panic because we're overriding it.